waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -67,16 +63,42 @@ genJobs spec = return $ genJob <$> getJobs spec
                   [ "jobName" .= jobName,
                     "jobPerformFnName" .= jobPerformFnName,
                     "jobPerformFnImportStatement" .= jobPerformFnImportStatement,
-                    "jobFactoryName" .= show (jobFactoryForJob job)
+                    "jobExecutorFilename" .= toFilePath (basename $ jobCreatorFilePath $ executor job),
+                    "jobPerformOptions" .= show (fromMaybe AS.JSON.emptyObject (options . perform $ job))
                   ]
             )
 
-genJobFactories :: Generator [FileDraft]
-genJobFactories = return $ genJobFactory <$> jobFactories
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+genJobExecutors :: Generator [FileDraft]
+genJobExecutors = return $ jobExecutorFds ++ jobExecutorHelperFds
   where
-    genJobFactory :: JobFactory -> FileDraft
-    genJobFactory jobFactory =
-      let jobFactoryFp = jobFactoryFilePath jobFactory
-          sourceTemplateFp = C.asTmplFile jobFactoryFp
-          destinationServerFp = C.asServerFile jobFactoryFp
-       in C.mkTmplFdWithDstAndData sourceTemplateFp destinationServerFp Nothing
+    jobExecutorFds :: [FileDraft]
+    jobExecutorFds = genJobExecutor <$> jobExecutors
+
+    genJobExecutor :: JobExecutor -> FileDraft
+    genJobExecutor jobExecutor =
+      let jobExecutorFp = jobCreatorFilePath jobExecutor
+          sourceTemplateFp = C.asTmplFile jobExecutorFp
+       in C.mkTmplFd sourceTemplateFp
+
+    jobExecutorHelperFds :: [FileDraft]
+    jobExecutorHelperFds =
+      [ C.mkTmplFd $ C.asTmplFile [relfile|src/jobs/pgBoss.js|],
+        C.mkTmplFd $ C.asTmplFile [relfile|src/jobs/SubmittedJob.js|]
+      ]
+
+jobCreatorFilePath :: JobExecutor -> Path' (Rel d) File'
+jobCreatorFilePath Passthrough = [relfile|src/jobs/passthroughJob.js|]
+jobCreatorFilePath PgBoss = [relfile|src/jobs/pgBossJob.js|]
+
+pgBossVersionBounds :: String
+pgBossVersionBounds = "^7.2.1"
```

Comment:
I use it below in the context of a `AS.Dependency.Dependency` instead, as that is what is used in:
```haskell
N.NpmDepsForWasp
    { N.waspDependencies = <here>
```

Reply:
Yeah, we decided to leave the `AS.Dependency` as a string for now since changing it would bring many complications, but we are planning to change that in the future.

You can still define the dependency version and its bounds in terms of `SemanticVersion` though (in the same way it's used in Wasp/Generator/Common.hs):
```haskell
pgBossVersionBounds :: SV.VersionBounds
pgBossVersionBounds = SV.BackwardsCompatibleWith (SV.Version 7 2 1)

pgBossDependency :: AS.Dependency.Dependency
pgBossDependency = AS.Dependency.make ("pg-boss", show pgBossVersionBounds)
```
The only benefit you'd currently get is from abstracting an arbitrary symbol (i.e., `^`) behind an explanation (i.e., `CompabileWith`.

Also, I like the way you kept all job-related dependencies in the same place. It's probably a good idea to do the same thing for other future dependencies (and possibly current dependencies, which are currently all grouped inside Common.hs).

Reply:
Nice, I like it and will use here and moving forward. Thanks! 👍🏻